tango_simlib/tests/test_dish.py

```diff
@@ -320,3 +323,60 @@ def test_slew(self):
             self.assertEqual(pointing_state_quant.last_val,
                              pointing_state_enum_labels.index('READY'))
+
+    def test_long_running(self):
+        with mock.patch('time.sleep') as sleep_mock:
+            # We use the value '4.5' as an arbitrary value here
+            self.model.sim_actions['LongRun'](4.5)
+            # We use value '5' as the time the long_running cmd is going sleep for.
+            calls = [call(5)]
+            num_method_calls = sleep_mock.call_count
+            self.assertEquals(num_method_calls, 1, "The mocked method 'sleep' was called"
+                              " more than once.")
+            self.assertEquals(calls, sleep_mock.mock_calls, "The calls made to the"
+                              " mocked method 'sleep' are not the same.")
+
+
+
+class test_Device(ClassCleanupUnittestMixin, unittest.TestCase):
+
+    longMessage = True
+
+    @classmethod
+    def setUpClassWithCleanup(cls):
+        cls.tango_db = cleanup_tempfile(cls, prefix='tango', suffix='.db')
+        cls.data_descr_files = []
+        cls.data_descr_files.append(
+            pkg_resources.resource_filename('tango_simlib.tests',
+                                            'DishElementMaster.xmi'))
+        cls.data_descr_files.append(pkg_resources.resource_filename(
+            'tango_simlib.tests', 'DishElementMaster_SIMDD.json'))
+        cls.device_name = 'test/nodb/tangodeviceserver'
+        model = tango_sim_generator.configure_device_model(cls.data_descr_files,
+                                                           cls.device_name)
+        cls.TangoDeviceServer = tango_sim_generator.get_tango_device_server(
+                model, cls.data_descr_files)[0]
+        cls.tango_context = TangoTestContext(cls.TangoDeviceServer,
+                                             device_name=cls.device_name,
+                                             db=cls.tango_db)
+        start_thread_with_cleanup(cls, cls.tango_context)
+
+    def setUp(self):
+        super(test_Device, self).setUp()
+        self.device = self.tango_context.device
+
+    def test_long_running(self):
+        """Testing the device's long running command."""
+        long_running_cmd_min_exec_time = 5
+        expected_runtime = 5.00
+        self.device.set_timeout_millis(5100)
+        initial_time = time.time()
+        # We use the value '4.5' as an arbitrary value here as the cmd is expecting
+        # an input of type DevFloat.
+        self.device.command_inout('LongRun', 4.5)
+        final_time = time.time()
+        actual_runtime = final_time - initial_time
+        self.assertGreater(actual_runtime, long_running_cmd_min_exec_time,
+                           "The long-running command 'LongRun' did not run for as long"
+                           " as expected.")
+        self.assertAlmostEqual(expected_runtime, actual_runtime, delta=0.01)
```

Comment:
These two asserts are checking almost exactly the same thing.  The second version is a little more robust as it has a tolerance on the time.   Anyway, it works with both, so we can leave it as is.